experimental/dds/tree2/src/shared-tree/sharedTree.ts

```diff
@@ -126,6 +129,26 @@ export interface ISharedTree extends ISharedObject, TypedTreeChannel {
 	schematizeView<TRoot extends TreeFieldSchema>(
 		config: InitializeAndSchematizeConfiguration<TRoot>,
 	): ISharedTreeView;
+
+	/**
+	 * Like {@link ISharedTree.schematizeView}, but will never modify the document.
+	 *
+	 * @param schema - The view schema to use.
+	 * @param onSchemaIncompatible - A callback.
+	 * Invoked when the returned ISharedTreeView becomes invalid to use due to a change to the stored schema which makes it incompatible with the view schema.
+	 * Called at most once.
+	 * @returns a view compatible with the provided schema, or undefined if the stored schema is not compatible with the provided view schema.
+	 * If this becomes invalid to use due to a change in the stored schema, onSchemaIncompatible will be invoked.
+	 *
+	 * @privateRemarks
+	 * TODO:
+	 * Once views actually have a view schema, onSchemaIncompatible can become an event on the view (which ends its lifetime),
+	 * instead of a separate callback.
+	 */
+	requireSchema<TRoot extends TreeFieldSchema>(
```

Comment:
Long term, do you have any vague plans to maybe merge this method with `schematizeView`? They are pretty similar, and it's a little confusing to have both. Fine for now, but I'm hoping we can make it simpler in the future when more of this has settled.

Reply:
I don't know where we will end up long term. Eventually we will want to make it possible to have schematize failing be recoverable (like it is for requireSchema), but they might also diverge further when we start allowing schematize to do lay editing with adapters for schema evolution.